ext/storage_sources/s3_store/s3_connection.cpp

```diff
@@ -33,6 +39,11 @@ S3Connection::S3Connection(const Aws::S3Crt::ClientConfiguration &config,
   const std::string &bucketName, const std::string &objPrefix)
     : _s3CrtClient(config), _bucketName(bucketName), _objectPrefix(objPrefix)
 {
+    /* Confirm that we can access the bucket, else fail. */
+    bool exists;
+    int ret = BucketExists(exists);
+    if (ret != 0 || !exists)
+        throw std::invalid_argument(_bucketName + " : No such bucket.");
```

Comment:
Same, we can add more info.

Reply:
As mentioned above, we will resolve this as a general problem through WT-8855.